components/mediators/cache/org.wso2.carbon.mediator.cache/src/main/java/org/wso2/carbon/mediator/cache/CacheMediatorFactory.java

```diff
@@ -207,6 +217,29 @@ protected Mediator createSpecificMediator(OMElement elem, Properties properties)
                             } else {
                                 cache.setResponseCodes(CachingConstants.ANY_RESPONSE_CODE);
                             }
+
+                            OMElement enableCacheControlElem =
+                                    protocolElem.getFirstChildWithName(ENABLE_CACHE_CONTROL_Q);
+                            if (enableCacheControlElem != null) {
+                                String cacheControlElemText = enableCacheControlElem.getText();
+                                if (!"".equals(cacheControlElemText) && cacheControlElemText != null) {
```

Comment:
You can check this as `if(cacheControlElemText != null && cacheControlElemText.isEmpty())`